### PR TITLE
AUTH-482: set required-scc for openshift workloads

### PR DESCRIPTION
**Pull request by liouk**
Workloads affected:
- `openshift-ingress/router-default` deployment
- `openshift-ingress-canary/ingress-canary` daemonset
- `openshift-ingress-operator/ingress-operator` deployment

### CODE REVIEW

File: pkg/manifests/assets/router/deployment.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ spec:
     metadata:
       annotations:
         target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+        openshift.io/required-scc: hostnetwork
```
Comment (Miciah) [inline pkg/manifests/assets/router/deployment.yaml:12]: The deployment may use "hostnetwork" or "restricted", depending on the ingresscontroller configuration (specifically `IngressController.spec.endpointPublishingStrategy.type`).  So I think we will need `desiredRouterDeployment` in `pkg/operator/controller/ingress/deployment.go` to set this annotation at run-time —right?

By the way, this PR reminds me of some related PRs, #743 and #981, which updated the security context based on previous customer cases where we observed that the wrong SCC could be selected.

Comment (liouk) [reply pkg/manifests/assets/router/deployment.yaml]: Thanks for the pointer @Miciah :slightly_smiling_face: I've pushed a change that addresses this.

**Comment by liouk** [comment]
/retest

failures seem to be infra/connectivity related

File: pkg/manifests/assets/router/deployment.yaml (hunk 1 above)
Comment (Miciah) [reply pkg/manifests/assets/router/deployment.yaml]: You're right, it seems that we missed this change [from the OCP 4.11 release notes](https://docs.openshift.com/container-platform/4.11/release_notes/ocp-4-11-release-notes.html#ocp-4-11-auth-pod-security-admission):

> The `restricted` SCC is no longer available to users of new clusters, unless the access is explicitly granted. In clusters originally installed in OpenShift Container Platform 4.10 or earlier, all authenticated users can use the `restricted` SCC when upgrading to OpenShift Container Platform 4.11 and later.  

And observations of CI artifacts confirm this:

- periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-aws-ovn-upgrade has `openshift.io/scc: restricted`.
- periodic-ci-openshift-release-master-ci-4.10-e2e-aws-serial has `openshift.io/scc: restricted`.
- periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade has `openshift.io/scc: restricted`.
- periodic-ci-openshift-release-master-ci-4.11-e2e-aws-serial has `openshift.io/scc: hostnetwork`.
- periodic-ci-openshift-release-master-ci-4.12-upgrade-from-stable-4.11-e2e-aws-ovn-upgrade has `openshift.io/scc: hostnetwork`.
- periodic-ci-openshift-release-master-ci-4.12-e2e-aws-sdn-serial has `openshift.io/scc: hostnetwork`.

The intention is definitely that we use "restricted" when we don't need "hostnetwork".  I've filed [OCPBUGS-34373](https://issues.redhat.com/browse/OCPBUGS-34373) and posted #1064 to address this oversight.  Thanks for pointing it out!

Comment (liouk) [reply pkg/manifests/assets/router/deployment.yaml]: Glad this helped :) I'll wait for #1064 to merge and then rebase this and try again :+1:

**Comment by liouk** [comment]
@candita I've retried `e2e-aws-gatewayapi` a couple of times and it seems to be quite flaky, every time different tests fail. Is there something to be done, other than retrying until it succeeds? Failures don't seem related to the changes of this PR.

**Comment by candita** [comment]
> I've retried `e2e-aws-gatewayapi` a couple of times and it seems to be quite flaky, every time different tests fail. Is there something to be done, other than retrying until it succeeds? Failures don't seem related to the changes of this PR.

@liouk the last two retries failed with a cluster installation error, which has nothing to do with this test or your PR:

> level=fatal msg=failed to fetch Cluster: failed to fetch dependency of "Cluster": failed to generate asset "Platform Permissions Check": validate AWS credentials: current credentials insufficient for performing cluster installation
> Installer exit with code 1
> 

The other failing tests ("no such host") illustrate an AWS DNS propagation delay that we're investigating.  It comes and goes.  `e2e-aws-gatewayapi` isn't required, so it's ok to proceed.

File: pkg/operator/controller/ingress/deployment_test.go [hunk 2]
```diff
@@ -502,6 +503,12 @@ func TestDesiredRouterDeploymentSpecTemplate(t *testing.T) {
 		t.Errorf("expected annotation %q to be %q, got %q", WorkloadPartitioningManagement, expected, val)
 	}
 
+	if val, ok := deployment.Spec.Template.Annotations[securityv1.RequiredSCCAnnotation]; !ok {
+		t.Errorf("missing annotation %q", securityv1.RequiredSCCAnnotation)
+	} else if expected := "hostnetwork"; expected != val {
+		t.Errorf("expected annotation %q to be %q, got %q", securityv1.RequiredSCCAnnotation, expected, val)
+	}
+
```
Comment (Miciah) [inline pkg/operator/controller/ingress/deployment_test.go:515]: These changes belong in the next commit.

**Comment by Miciah** [comment]
We've been trying to be more consistent in commit messages and follow [upstream guidelines](https://www.kubernetes.dev/docs/guide/pull-requests/#commit-message-guidelines).  Would you mind adding a reference to https://issues.redhat.com/browse/AUTH-482 to the commit messages and making sure lines are wrapped (50 chars for the subject, 72 for the body)?

**Comment by liouk** [comment]
Thanks for the correction, @Miciah. I've fixed the commits accordingly, please let me know if they match your expectations.

**Comment by candita** [comment]
failed to wait for importing imagestreamtags on ci-op-h92w2wlz/pipeline:origin-centos-8: failed to import tag(s) [origin-centos-8] on image stream ci-op-h92w2wlz/pipeline because of missing definition in the spec

/retest

**Comment by Miciah** [comment]
Thanks!
/lgtm

**Pull request merged**